src/System.Diagnostics.Process/src/System/Diagnostics/ProcessWaitState.Unix.cs

```diff
@@ -90,28 +90,48 @@ public void Dispose()
         }
 
         /// <summary>
-        /// Global table that maps process IDs to the associated shared wait state information.
+        /// Global table that maps process IDs of non-child Processes to the associated shared wait state information.
         /// </summary>
         private static readonly Dictionary<int, ProcessWaitState> s_processWaitStates =
             new Dictionary<int, ProcessWaitState>();
 
+        /// <summary>
+        /// Global table that maps process IDs of child Processes to the associated shared wait state information.
+        /// </summary>
+        private static readonly Dictionary<int, ProcessWaitState> s_childProcessWaitStates =
+            new Dictionary<int, ProcessWaitState>();
+
         /// <summary>
         /// Ensures that the mapping table contains an entry for the process ID,
         /// increments its ref count, and returns it.
         /// </summary>
         /// <param name="processId">The process ID for which we need wait state.</param>
         /// <returns>The wait state object.</returns>
-        internal static ProcessWaitState AddRef(int processId)
+        internal static ProcessWaitState AddRef(int processId, bool isChild)
         {
-            lock (s_processWaitStates)
+            lock (s_childProcessWaitStates)
             {
                 ProcessWaitState pws;
-                if (!s_processWaitStates.TryGetValue(processId, out pws))
+                if (isChild)
                 {
-                    pws = new ProcessWaitState(processId);
-                    s_processWaitStates.Add(processId, pws);
+                    pws = new ProcessWaitState(processId, isChild);
+                    s_childProcessWaitStates.Add(processId, pws);
+                    pws._outstandingRefCount++; // For Holder
+                    pws._outstandingRefCount++; // Decremented in CheckChildren
+                }
+                else
+                {
+                    lock (s_processWaitStates)
+                    {
+                        if (!s_childProcessWaitStates.TryGetValue(processId, out pws)
+                        && !s_processWaitStates.TryGetValue(processId, out pws))
```

Comment:
Nit: for consistency, the && should be at the end of the previous line and this line indented